### ISSUE TITLE
[Feature]: Upload an album

### ISSUE THREAD

**Issue by ngugcx** [immich-app/immich#886]
### Feature detail

I have many photos on my computer, they are already arranged into albums in different directories.
It will be convenient if I can upload the whole album to immich directly instead of having to create it after uploading photos.

This feature request can apply to mobile app too.

### Platform

Web

**Comment by jrasm91** [comment]
I added something similar to the asset detail page, it probably would not require much work to add it here now. I added a reusable album selector modal that does most of the work.

**Comment by ngugcx** [comment]
> For the first feature "Upload an album" your can use the CLI to do that https://github.com/immich-app/CLI

Great. It would be even better to add it on the web.
How about add an 'Upload' button to the album detail page?
![image](https://user-images.githubusercontent.com/44153489/200974630-9f7b502e-e465-47dd-8c90-d132dc717597.png)

**Comment by ngugcx** [comment]
> I added something similar to the asset detail page, it probably would not require much work to add it here now. I added a reusable album selector modal that does most of the work.

Tried new version, it improves a lot. Great job!
Is there a simple to know which albums a photo belongs to in the "Photos" page?

**Comment by jrasm91** [comment]
> Is there a simple way to know which albums a photo belongs to in the "Photos" page?

I believe if you click the info button to open the side panel on the asset detail it'll show what albums the asset is in. Not sure if that's what you are talking about.

**Comment by jrasm91** [comment]
Are there still missing features for this issue? I think both options might be available now (1) upload and create albums via cli and (2) create albums from selected photos. (At least in the web client)

**Comment by ngugcx** [comment]
> How about add an 'Upload' button to the album detail page?
> ![image](https://user-images.githubusercontent.com/44153489/200974630-9f7b502e-e465-47dd-8c90-d132dc717597.png)

https://github.com/immich-app/immich/issues/886#issuecomment-1309621690

**Comment by jrasm91** [comment]
OK, that seems doable, I think we already have an "upload from computer" button on the normal photos page, so we should be able to add it on this page as well.

**Comment by ngugcx** [comment]
To avoid misunderstanding, the photos uploaded in this page are automatically added to current album.

**Comment by jrasm91** [comment]
Yeah, I understand and it shouldn't be too hard to implement since we already have code that allows uploading from the computer, so we should be able to tweak it (for the album page) to both upload and add to the selected album.

**Comment by ngugcx** [comment]
> Yeah, I understand and it shouldn't be too hard to implement since we already have code that allows uploading from the computer, so we should be able to tweak it (for the album page) to both upload and add to the selected album.

I checked it again, we need an 'Upload' button in the create album page too. Is it possible to upload a directory? Then we can use the directory name as album name by default.
![image](https://user-images.githubusercontent.com/44153489/201827823-df82d257-06f5-4f92-9f7d-0c7a7699e6f5.png)

**Comment by alextran1502** [comment]
> > Yeah, I understand and it shouldn't be too hard to implement since we already have code that allows uploading from the computer, so we should be able to tweak it (for the album page) to both upload and add to the selected album.
> 
> I checked it again, is it possible to upload a directory? Then we can use the directory name as album name by default when creating new album.
> 
> When creating new album, where to put the 'Upload' button? ![image](https://user-images.githubusercontent.com/44153489/201829733-0f8fb547-56c1-47f5-84f3-17c5baeb5964.png)

I would keep this simple for ease of maintenance, adding the upload from the computer button makes sense, you can manually enter the album name afterward
